### ISSUE TITLE
Unable to Export Default Solution

### ISSUE THREAD

**Issue by albernhagen** [seanmcne/Microsoft.Xrm.Data.PowerShell#137]
Hello,

I'm working on a process for us to compare the default solutions of multiple environments to find field inconsistencies and other issues that cause our deployments to fail.  When using the Export-CrmSolution cmdlet to export the solution with name 'Default', I get the below error.  Any ideas?

Thanks!

```
Mode                LastWriteTime     Length Name
----                -------------     ------ ----
d----          8/4/2016   2:27 PM            solutions
Export-CrmSolution : System.Management.Automation.MethodInvocationException: Exception calling "WriteAllBytes" with "2" argument(s): "Value cannot be null.
Parameter name: bytes" ---> System.ArgumentNullException: Value cannot be null.
Parameter name: bytes
   at System.IO.File.WriteAllBytes(String path, Byte[] bytes)
   at CallSite.Target(Closure , CallSite , RuntimeType , Object , Object )
   --- End of inner exception stack trace ---
   at System.Management.Automation.ExceptionHandlingOps.CheckActionPreference(FunctionContext funcContext, Exception exception)
   at System.Management.Automation.Interpreter.ActionCallInstruction`2.Run(InterpretedFrame frame)
   at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
   at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
At C:\VSWorkspaces\CRM\Mainline\DeploymentScripts\DownloadSolution.ps1:50 char:2
+     Export-CrmSolution -conn $conn -SolutionName $solutionName -SolutionFilePath $s ...
+    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
    + FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException,Export-CrmSolution
```

**Comment by kenakamu** [comment]
I will debug it for next iteration.

**Comment by carfup** [comment]
Stupid question but it's not normal that we can't export the default solution and this explain the possibility to split everything is several solutions ?

**Comment by seanmcne** [comment]
@carfup  can you rephrase, I don't follow the question?

**Comment by carfup** [comment]
My bad, forget about my last entry, i was totally wrong ... You can delete it.

**Comment by tjahnks23** [comment]
@kenakamu has there been a solution for the WriteAllBytes() error?

**Comment by seanmcne** [comment]
Yes I believe #184 will taks care of it - if the response is null it should throw instead of return a null value to write.  Have you tried downloading this latest repo to try and see if the issue still exists?

https://github.com/seanmcne/Microsoft.Xrm.Data.PowerShell/commit/04e4e66a9ee3f4a34bb7fc9b8247961ea535920a

**Comment by tjahnks23** [comment]
@seanmcne thanks for the quick reply - I'll need to get that and let you know!
